sdk/resourcemanager/Azure.ResourceManager.Core/src/SingletonOperationsBase.cs

```diff
@@ -18,14 +20,19 @@ protected SingletonOperationsBase()
         /// <summary>
         /// Initializes a new instance of the <see cref="SingletonOperationsBase"/> class.
         /// </summary>
-        /// <param name="clientContext"></param>
-        /// <param name="id"></param>
-        internal SingletonOperationsBase(ClientContext clientContext, ResourceIdentifier id)
-            : base(clientContext, ResourceIdentifier.RootResourceIdentifier)
+        /// <param name="parent"></param>
+        internal SingletonOperationsBase(OperationsBase parent)
+            : base(new ClientContext(parent.ClientOptions, parent.Credential, parent.BaseUri, parent.Pipeline), ResourceIdentifier.RootResourceIdentifier)
         {
-            ParentId = id;
+            Parent = parent;
+            ParentId = parent.Id;
         }
 
+        /// <summary>
+        /// Gets the parent resource of this resource.
+        /// </summary>
+        protected OperationsBase Parent { get; set; }
```

Comment:
I think this can be just a getter no setter since we don't expect this to be set outside the constructor